axelrod/strategies/titfortat.py

```diff
@@ -772,3 +774,52 @@ def strategy(self, opponent: Player) -> Action:
             self.retaliate_count -= 1
             return D
         return C
+
+
+@FinalTransformer((D,), name_prefix=None)
+class Michaelos(Player):
+    """
+    Plays similar to Tit-for-Tat with two exceptions:
+    1) Defect on last turn.
+    2) After own defection and opponent's cooperation, 50 percent of the time,
+    cooperate. The other 50 percent of the time, always defect for the rest of
+    the game.
+
+    Names:
+
+    - Michaelos: [LessWrong2011]_
+    """
+
+    name = 'Michaelos'
+    classifier = {
+        'memory_depth': 1,
```

Comment:
I believe that the memory depth is not 1.

If on turn `n` Michaelos defects and the opponent cooperates then 50% of the time I defect and defect throughout, However if I have already **become** a defector than I will defect 100% of the time so I not only need to know the previous turns (that I D and you C but I also need to know if I D'd because I have already "transformed" in to a defector).

I believe the memory depth is in fact `float('inf')`:

If our actions are `...(D, D), (D, C)` then I do not know if my previous plays of `D` are TfT retaliations **or** because I have previously been turned in to a defector so I do not know if I need to randomly sample as a result of the last round being `(D, C)`.